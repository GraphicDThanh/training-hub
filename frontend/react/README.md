# React Redux Training
[Plan doc](https://docs.google.com/document/d/1lYywN0RIhrwsyFZ8uPZuaN5QM78mYuXFD608vS8YEbM/edit?pli=1#)
## ES6
## React Basic
### Practice 1 - examples React
## React Advanced
### Practice 2
## Redux Basic
## Redux Advanced
## Redux Recipes
