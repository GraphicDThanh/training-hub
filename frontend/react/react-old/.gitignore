node_modules
out
dist
.DS_Store
.git
.docpad.db
vendor/
.vagrant
bower-components/
bower_components/
npm-debug.log
.gitignore~
vagrant/Vagrantfile~
build/
