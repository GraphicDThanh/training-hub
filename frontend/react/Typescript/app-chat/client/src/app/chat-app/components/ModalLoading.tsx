import * as React from 'react';
import Modal from '../../../common/components/Modal';

interface Props {
  message?: any[];
}

const ModalLoading = ({ message }: Props) => {
  if (message) {
    message = message.map((content: any) => content.split('_').join(' '));
  }

  let props = {
    title: 'The app on loading ...'
  };

  return <Modal {...props}>
     <div className="loading-wrapper">
      <div className="wrapper-loader">
        <div className="loader">Loading...</div>
      </div>
      {
        message && message.length &&
        <div className="message">{message.join(', ')}</div>
      }
    </div>
  </Modal>
}

export default ModalLoading;