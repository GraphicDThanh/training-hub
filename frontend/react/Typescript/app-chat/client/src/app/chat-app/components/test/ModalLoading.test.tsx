import * as React from 'react';
import { render } from 'enzyme';
import ModalLoading from '../ModalLoading';

describe('component Modal Loading', () => {
  it('snapshot', () => {
    let message = ['ACTION_1', 'ACTION_2'];
    const renderComponent = render(<ModalLoading message={message}/>);
    expect(renderComponent).toMatchSnapshot();
  });

  it('snapshot without props', () => {
    const renderComponent = render(<ModalLoading/>);
    expect(renderComponent).toMatchSnapshot();
  });
})