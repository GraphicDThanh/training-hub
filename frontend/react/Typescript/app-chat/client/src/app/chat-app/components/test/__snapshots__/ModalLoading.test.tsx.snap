// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component Modal Loading snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "modal-wrapper",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "model-dialog",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "modal-content",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "modal-header",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "title",
                    },
                    "children": Array [
                      Object {
                        "data": "The app on loading ...",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "h4",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "modal-body",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "loading-wrapper",
                      },
                      "children": Array [
                        Object {
                          "attribs": Object {
                            "class": "wrapper-loader",
                          },
                          "children": Array [
                            Object {
                              "attribs": Object {
                                "class": "loader",
                              },
                              "children": Array [
                                Object {
                                  "data": "Loading...",
                                  "next": null,
                                  "parent": [Circular],
                                  "prev": null,
                                  "type": "text",
                                },
                              ],
                              "name": "div",
                              "namespace": "http://www.w3.org/1999/xhtml",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "tag",
                              "x-attribsNamespace": Object {
                                "class": undefined,
                              },
                              "x-attribsPrefix": Object {
                                "class": undefined,
                              },
                            },
                          ],
                          "name": "div",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": Object {
                            "attribs": Object {
                              "class": "message",
                            },
                            "children": Array [
                              Object {
                                "data": "ACTION 1, ACTION 2",
                                "next": null,
                                "parent": [Circular],
                                "prev": null,
                                "type": "text",
                              },
                            ],
                            "name": "div",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": null,
                            "parent": [Circular],
                            "prev": [Circular],
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "class": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "class": undefined,
                            },
                          },
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "class": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "class": undefined,
                          },
                        },
                        Object {
                          "attribs": Object {
                            "class": "message",
                          },
                          "children": Array [
                            Object {
                              "data": "ACTION 1, ACTION 2",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "text",
                            },
                          ],
                          "name": "div",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": null,
                          "parent": [Circular],
                          "prev": Object {
                            "attribs": Object {
                              "class": "wrapper-loader",
                            },
                            "children": Array [
                              Object {
                                "attribs": Object {
                                  "class": "loader",
                                },
                                "children": Array [
                                  Object {
                                    "data": "Loading...",
                                    "next": null,
                                    "parent": [Circular],
                                    "prev": null,
                                    "type": "text",
                                  },
                                ],
                                "name": "div",
                                "namespace": "http://www.w3.org/1999/xhtml",
                                "next": null,
                                "parent": [Circular],
                                "prev": null,
                                "type": "tag",
                                "x-attribsNamespace": Object {
                                  "class": undefined,
                                },
                                "x-attribsPrefix": Object {
                                  "class": undefined,
                                },
                              },
                            ],
                            "name": "div",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": [Circular],
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "class": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "class": undefined,
                            },
                          },
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "class": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "class": undefined,
                          },
                        },
                      ],
                      "name": "div",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "class": "modal-body",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "loading-wrapper",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "class": "wrapper-loader",
                        },
                        "children": Array [
                          Object {
                            "attribs": Object {
                              "class": "loader",
                            },
                            "children": Array [
                              Object {
                                "data": "Loading...",
                                "next": null,
                                "parent": [Circular],
                                "prev": null,
                                "type": "text",
                              },
                            ],
                            "name": "div",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "class": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "class": undefined,
                            },
                          },
                        ],
                        "name": "div",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": Object {
                          "attribs": Object {
                            "class": "message",
                          },
                          "children": Array [
                            Object {
                              "data": "ACTION 1, ACTION 2",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "text",
                            },
                          ],
                          "name": "div",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": null,
                          "parent": [Circular],
                          "prev": [Circular],
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "class": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "class": undefined,
                          },
                        },
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                        },
                      },
                      Object {
                        "attribs": Object {
                          "class": "message",
                        },
                        "children": Array [
                          Object {
                            "data": "ACTION 1, ACTION 2",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "div",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": Object {
                          "attribs": Object {
                            "class": "wrapper-loader",
                          },
                          "children": Array [
                            Object {
                              "attribs": Object {
                                "class": "loader",
                              },
                              "children": Array [
                                Object {
                                  "data": "Loading...",
                                  "next": null,
                                  "parent": [Circular],
                                  "prev": null,
                                  "type": "text",
                                },
                              ],
                              "name": "div",
                              "namespace": "http://www.w3.org/1999/xhtml",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "tag",
                              "x-attribsNamespace": Object {
                                "class": undefined,
                              },
                              "x-attribsPrefix": Object {
                                "class": undefined,
                              },
                            },
                          ],
                          "name": "div",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": [Circular],
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "class": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "class": undefined,
                          },
                        },
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                        },
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "modal-header",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "title",
                      },
                      "children": Array [
                        Object {
                          "data": "The app on loading ...",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "h4",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`component Modal Loading snapshot without props 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "modal-wrapper",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "model-dialog",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "modal-content",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "modal-header",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "title",
                    },
                    "children": Array [
                      Object {
                        "data": "The app on loading ...",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "h4",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "modal-body",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "loading-wrapper",
                      },
                      "children": Array [
                        Object {
                          "attribs": Object {
                            "class": "wrapper-loader",
                          },
                          "children": Array [
                            Object {
                              "attribs": Object {
                                "class": "loader",
                              },
                              "children": Array [
                                Object {
                                  "data": "Loading...",
                                  "next": null,
                                  "parent": [Circular],
                                  "prev": null,
                                  "type": "text",
                                },
                              ],
                              "name": "div",
                              "namespace": "http://www.w3.org/1999/xhtml",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "tag",
                              "x-attribsNamespace": Object {
                                "class": undefined,
                              },
                              "x-attribsPrefix": Object {
                                "class": undefined,
                              },
                            },
                          ],
                          "name": "div",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "class": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "class": undefined,
                          },
                        },
                      ],
                      "name": "div",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "class": "modal-body",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "loading-wrapper",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "class": "wrapper-loader",
                        },
                        "children": Array [
                          Object {
                            "attribs": Object {
                              "class": "loader",
                            },
                            "children": Array [
                              Object {
                                "data": "Loading...",
                                "next": null,
                                "parent": [Circular],
                                "prev": null,
                                "type": "text",
                              },
                            ],
                            "name": "div",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "class": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "class": undefined,
                            },
                          },
                        ],
                        "name": "div",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                        },
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "modal-header",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "title",
                      },
                      "children": Array [
                        Object {
                          "data": "The app on loading ...",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "h4",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
